WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -30,7 +30,7 @@ public partial class CoinJoinStateViewModel : ViewModelBase
 	private const string WaitingRoundMessage = "Waiting for a round";
 	private const string PlebStopMessage = "Coinjoining might be uneconomical";
 	private const string PlebStopMessageBelow = "Receive more funds or press play to bypass";
-	private const string WaitingForConfirmedFundsMessage = "Waiting for confirmed funds";
+	private const string NoCoinsToMixMessage = "No funds available to mix.";
```

Comment:
IMO this message is not helpful but causes confusion. How about not displaying a detail like this, but just simply saying in general that `Waiting for a round`.

@molnard What do you think about the idea of whenever there is an error and the client cannot join a round, instead of displaying the actual error, do not bother the users with any error, but display "Waiting for a round" message.
The log would still contain the details. but users wouldn't be confused.

Reply:
> How about not displaying a detail like this, but just simply saying in general that `Waiting for a round`.

The users might think that no round has happened during all that time that they are waiting.

Reply:
I can imagine it like this:

- If any error happened before input reg phase, we don't have to tell it to the user, we just need to keep displaying the `Waiting for a round` message.
- If any error happened once the client joined a round, I wouldn't display a detailed message, just `Round finished unsuccessfully`.

With this, the users will be bothered less and also they won't get confusing messages and devs are still able to get details messages from logs.

Reply:
Such messages open the door for users being confused and requesting support.
What would you say to users complaining that they have been running Wasabi for hours and they couldn't participate in coinjoin, and they only see `Waiting for a round`?

I guess `Not enough funds available to mix.` is better and more descriptive than `Waiting for a round`.

Reply:
> Such messages open the door for users being confused and requesting support. What would you say to users complaining that they have been running Wasabi for hours and they couldn't participate in coinjoin, and they only see `Waiting for a round`?
> 
> I guess `Not enough funds available to mix.` is better and more descriptive than `Waiting for a round`.

See the real reason for that error outlined here, this is not really true also: https://github.com/zkSNACKs/WalletWasabi/issues/10313#issuecomment-1547476397

1. So I see these can be either message ""No coin was selected from candidate coins" but no idea what real reason is or might have been reason "...Probably it was not economical"?
2. The message maybe should indicate "The coordinator rejected all inputs"or something more user friendly ?
3. Possible message "No candidate coins available to mix." for music box ?

Not idea how we can make one message that fits all 3 cases, that's why I proposed created 3 different errors so we can later display adequate messages to user.

Reply:
Yes I have seen your investigation, and to me it seems that whatever the reason is all those cases are about `Not enough funds available to mix.` either because coins are unconfirmed, or banned, or not economical, ...

Of course creating 3 different error messages is more precise and gives more information to the users but I think we don't want to go that road.

Reply:
It seems like there is no good solution...

`Not enough funds available to mix.`
IMO this is not good either, as the user only sees their balance, nothing else. Imagine the balance is 1 BTC and MusicBox says there are not enough funds to mix. The user would immediately think "Hey, I have 1 BTC, why does it say it?"

More messages? No, https://github.com/zkSNACKs/WalletWasabi/issues/8695.

My idea, https://github.com/zkSNACKs/WalletWasabi/pull/10697#discussion_r1201778195
nACK by @yahiheb.
Replying to this https://github.com/zkSNACKs/WalletWasabi/pull/10697#discussion_r1201804162:
Both solutions `Not enough funds available to mix.` and mine can cause confusion, but mine makes the music box to change state (visually) fewer times and also removes some messages which align with https://github.com/zkSNACKs/WalletWasabi/issues/8695.
Fewer messages = less information to confuse the user.
And also the user has nothing to do with those messages, they cannot fix it, and the only thing they can do is wait so the messages are unnecessary for the user.

Reply:
> What do you think about the idea of whenever there is an error and the client cannot join a round, instead of displaying the actual error, do not bother the users with any error, but display "Waiting for a round" message.
> The log would still contain the details. but users wouldn't be confused.

Music box is about to be refactored. I suggest to not spend much time on it. The less change the better.

Reply:
@soosr after some rough understanding, and also knowing that music box will have a refactoring, i would go for: "Not enough funds eligible to mix" its incomprehensible enough, either way user will have a hard time to understand the context in detail.